frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -291,6 +309,15 @@ pub mod pallet {
         NotActiveStaker,
         /// Transfering nomination to the same contract
         NominationTransferToSameContract,
+        /// Only staker or current delegate account can delegate reward
+        NotAuthorizedToDelegate,
+        /// Reward is already delegated to the account
+        AlreadyDelegated,
+        /// Can not set reward delegate if rewards are set to be re staked
+        RewardDestinationStakeBalance,
+        /// There is no reward delegate set
+        NoDelegateSet,
```

Comment:
Have you tested these out?

Reply:
No, I should have added a test for each event or error possible.